src/result.php

```diff
@@ -1,36 +1,54 @@
 <?php
-  header('Content-type: text/javascript');
   header('Cache-control: no-cache');
   require_once('/config/dbconfig.php');
+  $data = array();
+  function isVersionNumber($val) {
+    if (preg_match('/^[0-9.rc]+$/', $val)) { return true; }
+    return false;
+  }
+  function addRow(&$data, $key, $type, $count = 1) {
+    $keyToCol = array('good' => 0, 'nolike' => 1, 'rollback' => 2);
+    if (!isVersionNumber($key)) { return; }
+    if (!isset($data[$key])) { $data[$key] = array(0,0,0); }
+    $data[$key][$keyToCol[$type]] += $count;
+  }
+
+  function addIssue(&$data, $key, $issue, $count = 1) {
+    if (!isVersionNumber($key)) { return; }
+    // find the issue in the array already and increase value
+    for ($i = 3; $i < count($data[$key]); $i+=2) {
+      if ($data[$key][$i] == $issue) {
+        $data[$key][$i + 1] += $count;
+        return;
+      }
+    }
+    $data[$key][] = $issue;
+    $data[$key][] = $count;
+  }
+
   $db = @pg_connect("user=$dbuser password=$dbpass host=$dbserver dbname=$dbname");
   if (!$db) die('DB error');
-  $data = array();
-  $q = pg_query($db,
-    'select version, count(*) from jenkins_good group by version');
+  $q = pg_query($db, 'select version, count(*) from jenkins_good group by version');
   while ($row = pg_fetch_row($q)) {
-    $data[$row[0]] = array('good' => $row[1]);
+    addRow($data, $row[0], 'good', $row[1]);
   }
-  $q = pg_query($db,
-    'select version, rollback, issue from jenkins_bad order by version');
+  $q = pg_query($db, 'select version, rollback, issue from jenkins_bad order by version');
   while ($row = pg_fetch_row($q)) {
-    $data[$row[0]][$row[1]=='t' ? 'rollback' : 'nolike']++;
-    if ($row[2]) $data[$row[0]]['issues'][] = $row[2];
+    addRow($data, $row[0], $row[1]=='t' ? 'rollback' : 'nolike', $row[1]);
+    if ($row[2]) {
+      addIssue($data, $row[0], $row[2]);
+    }
   }
   pg_close($db);
-  print 'var data = {';
-  $first = true;
-  foreach ($data as $version => $info) {
-    if ($first) $first = false; else print ",\n";
-    foreach (array('good','rollback','nolike') as $key) if (!isset($info[$key])) $info[$key] = 0;
-    print json_encode($version) . ": [$info[good],$info[nolike],$info[rollback]";
-    if ($info[issues]) {
-      $issues = array_count_values($info['issues']);
-      arsort($issues);
-      foreach ($issues as $issue => $counts) {
-        print ",$issue,$counts";
-      }
```

Comment:
This was intentional, to sort most reported issues first in the list.

It's broken now:

> <img width="1099" alt="Screenshot 2021-12-28 at 09 54 00" src="https://user-images.githubusercontent.com/1831569/147548045-4acfc9e8-3faa-4d63-be37-e0b2c0694f7f.png">

Reply:
Filed https://issues.jenkins.io/browse/INFRA-3170

Reply:
https://github.com/jenkins-infra/jenkins.io/pull/4799

Reply:
Thanks!